libs/blocks/merch/merch.js

```diff
@@ -496,7 +497,13 @@ export async function openModal(e, url, offerType, hash, extraOptions) {
     const fragmentPath = url.split(/(hlx|aem).(page|live)/).pop();
     modal = await openFragmentModal(fragmentPath, getModal);
   } else {
-    modal = await openExternalModal(url, getModal, extraOptions);
+    const isThreeInOneModal = [MODAL_TYPE_3_IN_1.CRM, MODAL_TYPE_3_IN_1.D2P, MODAL_TYPE_3_IN_1.TWP].includes(el?.getAttribute('data-modal-type')) && el?.href;
```

Comment:
The constants are part of merch, but the code looks like it could be any type of modal, without a direct connection to merch. We should either a) make this more explicit, b) make the modal types more general so they can be used outside of `merch` scope.